ui/components/app/modals/snap-install-warning/snap-install-warning.js

```diff
@@ -0,0 +1,45 @@
+import React, { useCallback, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+import Modal from '../../modal';
+import CheckBox from '../../../ui/check-box/check-box.component';
+import { hideModal } from '../../../../store/actions';
+
+export default function SnapInstallWarning() {
+  const t = useI18nContext();
+  const dispatch = useDispatch();
+
+  const { onSubmit, snapName } = useSelector(
+    (state) => state.appState.modal.modalState.props,
+  );
+
+  const [isConfirmed, setIsConfirmed] = useState(false);
+
+  const onCheckboxClicked = useCallback(() => setIsConfirmed((i) => !i), []);
```

Comment:
Nit: We should use a more descriptive variable name here

```suggestion
  const onCheckboxClicked = useCallback(() => setIsConfirmed((confirmedState) => !confirmedState), []);
```